### ISSUE TITLE
M4B local download fails, renamed to MP3

### ISSUE THREAD

**Issue by cinvan2000** [advplyr/audiobookshelf-app#292]
### Steps to reproduce

1. Open Android App.
2. Find an audiobook in M4B-format. Click the "download" icon in the app, wait for it to reach 100%
3. On 100% it shows an Error instead. "Item download complete but failed to create library item"

### Expected behaviour
I would expect the file to show up in the local folder storage and the app able to play from that local storage.

### Actual behaviour
Similar to #154 but not the same problem i believe. The error is issued, in red, at the bottom of the screen. If you check local media within the app, the book isn't there. But if you do a manual scan the book **does** appear, but with the note _Not linked to server item_. Whats more, all the files has an .mp3 extension added. That is _Chapter 001.m4b_ becomes _Chapter 001.m4b.**mp3**_

Relevant lines from logcat:
```
07-17 20:05:54.454 26937 18514 D AbsDownloader: checkDownloads Download 001 - Chapter 001.m4b bytes 4892038 | bytes dled 4892038 | downloadStatus 8
07-17 20:05:54.454 26937 18514 D AbsDownloader: checkDownloads Download 001 - Chapter 001.m4b Done
07-17 20:05:54.456 26937 18514 D AbsDownloader: DOWNLOAD: Attempt move for file at destination file:///storage/emulated/0/Android/data/com.audiobookshelf.app/files/Download/001%20-%20Chapter%20001.m4b | Android/data/com.audiobookshelf.app/files/Download/001 - Chapter 001.m4b
07-17 20:05:54.456 26937 18514 D AbsDownloader: DOWNLOAD: Move file to final destination path: /storage/emulated/0/Audio/Audiobookshelf/The Long Earth/001 - Chapter 001.m4b
07-17 20:05:54.556 26937 26937 D AbsDownloader: DOWNLOAD: PREPARING MOVE FILE
07-17 20:05:54.675 26937 26937 D AbsDownloader: DOWNLOAD: FILE MOVE COMPLETED
07-17 20:05:54.676 26937 26937 D AbsDownloader: DOWNLOAD: COMPLETED FILE INFO /storage/emulated/0/Audio/Audiobookshelf/The Long Earth/001 - Chapter 001.m4b.mp3
```

At the end it gives me a null item id
`AbsDownloader: Item download complete The Long Earth | local library item id: null | Items remaining in Queue 0`

I also get one of these for every file
```
ModernMediaScanner: Trouble scanning /storage/emulated/0/Audio/Audiobookshelf/The Long Earth/001 - Chapter 001.m4b.mp3: java.lang.RuntimeException: setDataSource failed:
status = 0xFFFFFFEA
```

### Environment data

Audiobookshelf Version: v2.0.24 running in Docker.

- [X] Android App version 0.9.52-beta

#### Android Issue
Android version: 11

Device model: OnePlus6

Stock or customized system: Custom, LineageOS.

**Comment by wazabees** [comment]
Confirming I'm having the same issue. Google Pixel 6 Pro with Android 12. Server 2.0.24 running in docker and Android app v.0.9.52-beta.

**Comment by bskrtich** [comment]
I am also having this issue with the latest versions
Server: v2.1.0
App: Android 0.9.52-beta

**Comment by raviwarrier** [comment]
+1

**Comment by Benni-chan** [comment]
This also happens when downloading from the website using the m4b merge tool. I got a m4b.mp3 as a download

**Comment by cinvan2000** [comment]
Fixed for me with 0.9.53-beta and server 2.1.0, thanks!

**Comment by advplyr** [comment]
Fixed in [v0.9.53-beta](https://github.com/advplyr/audiobookshelf-app/releases/tag/v0.9.53-beta)
